libs/langchain_v1/langchain/agents/middleware/tool_call_limit.py

```diff
@@ -2,24 +2,36 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Annotated, Any, Literal
+from typing import TYPE_CHECKING, Annotated, Any, Generic, Literal
 
-from langchain_core.messages import AIMessage
+from langchain_core.messages import AIMessage, ToolMessage
 from langgraph.channels.untracked_value import UntrackedValue
+from langgraph.typing import ContextT
 from typing_extensions import NotRequired
 
 from langchain.agents.middleware.types import (
     AgentMiddleware,
     AgentState,
     PrivateStateAttr,
+    ResponseT,
     hook_config,
 )
 
 if TYPE_CHECKING:
     from langgraph.runtime import Runtime
 
+ExitBehavior = Literal["continue", "error", "end"]
+"""How to handle execution when tool call limits are exceeded.
 
-class ToolCallLimitState(AgentState):
+- `"continue"`: Block exceeded tools with error messages, let other tools continue (default)
+- `"error"`: Raise a `ToolCallLimitExceededError` exception
+- `"end"`: Stop execution immediately, injecting a ToolMessage and an AI message
+    for the single tool call that exceeded the limit. Raises `NotImplementedError`
```

Comment:
The doc-string is not precise for `end`

It's multiple pending tool calls or multiple parallel tool calls. and I'd probably explain "why" it's not implemented.